ci/parse-trivy-results.py

```diff
@@ -0,0 +1,121 @@
+"""Script to parse results from Trivy scan.
+
+1) Reads in results from two files (json and table format) generated by Trivy.
+2) Creates a GitHub issue in the given repo if any vulnerabilities were found.
+3) Fails the CI build if any vulerabilities of HIGH/CRITICAL severity exist.
+
+Both the json and table Trivy results must be created prior to running 
+this script. These can be generated with:
+    trivy image --format json --light --no-progress -o <OUTPUT_NAME> 
+        gcr.io/cre-tools/ts-bridge
+    trivy image --light  --no-progress -o <OUTPUT_NAME>
+        gcr.io/cre-tools/ts-bridge
+
+Usage:
+    python3 parse-trivy-results.py <TRIVY_FILE> <COMMIT_ID> <BUILD_ID> <FULL_TAG> <REPO_NAME> <TOKEN_FILE>
+
+"""
+import json
+import sys
+from github import Github
+
+CMDLINE_ARGS = 7
+
+def load_results():
+    """ Load the results from Trivy."""
+    TRIVY_OUT_JSON = "{}.json".format(sys.argv[1])
+    TRIVY_OUT_TABLE = "{}.table".format(sys.argv[1])
+    print(TRIVY_OUT_TABLE, TRIVY_OUT_JSON)
+    try:
+        with open(TRIVY_OUT_JSON) as f:
+            # Index is 0 because there is only one target built.
+            trivy_result = json.load(f)[0]
+        with open(TRIVY_OUT_TABLE) as f:
+            trivy_table = f.read()
+    except FileNotFoundError:
+        error = ("Please run \n"
+                "trivy image --format json --light --no-progress -o {} "
+                "gcr.io/cre-tools/ts-bridge \n"
+                "trivy image --light  --no-progress -o {} "
+                "gcr.io/cre-tools/ts-bridge \n"
+                ).format(TRIVY_OUT_JSON, TRIVY_OUT_TABLE)
+        sys.exit(error)
+    return [trivy_result, trivy_table]
+
+def get_severity_list(vulnerabilities):
+    """Filters out all the severities in a given list of vulnerabilities."""
+    severity_list = [v.get('Severity') for v in vulnerabilities]
+    return sorted(set(severity_list))
+
+def get_github_repo():
+    """Connects to GitHub API and returns a repo object."""
+    REPO_NAME = sys.argv[5]
+    with open(sys.argv[6]) as f:
+        token = f.read()
+    github = Github(token)
+    return github.get_repo(REPO_NAME)
+
+def create_issue(target_name, num_vulnerabilities, severity_list, table):
+    """Creates a Github issue with vulnerabilities as description."""
+    COMMIT_ID = sys.argv[2]
```

Comment:
This seems relatively unsafe parsing-wise as it's easy to mistake the order of a giant list of options and then there's no validation on the input.

- Please use `flags`.
- Please validate input. E.g. check if file exists, check if formatting is correct, etc.

Feel free to ping me to discuss.

Reply:
As discussed offline, I've added absl to check the cmdline args.